torch/ao/quantization/_dbr/module_swap_utils.py

```diff
@@ -58,6 +63,18 @@ def _swap_child_modules(
             if not type(mod) in dynamic_mappings:
                 continue
             reassign[local_fqn] = swap_module(mod, dynamic_mappings, {})
+        elif activation_int32_quantized:
+            # For now, only apply reference logic to modules quantized to
+            # int32. Do it automatically.
+            # TODO(future PR): extend this logic to more dtypes, and add
+            # the is_reference API flag instead of doing this automatically.
+            for orig_mod_type, _ref_mod_type in \
+                    DEFAULT_REFERENCE_STATIC_QUANT_MODULE_MAPPINGS.items():
+                if not isinstance(mod, orig_mod_type):  # type: ignore[arg-type]
```

Comment:
can we do this check in `swap_module`? not sure why we need to have a for loop here

Reply:
I see, looks like `swap_module` doesn't actually swap if the mapping is missing, maybe we can consider renaming it in the future.

Reply:
yeah renaming sounds good, it would make it clearer, like "maybe_swap_module", we need to redesign this api anyways to support swapping for multiple quantization modes anyways in the future I think, we can do it later